### PR TITLE
Work for release 0.19.x 839

### PR DESCRIPTION
**Pull request by sbrossie**

### CODE REVIEW

File: invoice/src/main/java/org/killbill/billing/invoice/usage/details/UsageConsumableInArrearTierUnitDetail.java [hunk 1]
```diff
@@ -70,4 +70,13 @@ public void setAmount(BigDecimal amount) {
...
         return tierBlockSize;
     }
+
+    public void updateQuantityAndAmount(final Integer targetQuantity) {
+        this.quantity = targetQuantity;
+        this.amount = computeAmount(tierPrice, targetQuantity);
+    }
+
+    private static BigDecimal computeAmount(final BigDecimal targetTierPrice, final Integer targetQuantity) {
+        return targetTierPrice.multiply(new BigDecimal(targetQuantity));
```
Comment (pierre) [inline invoice/src/main/java/org/killbill/billing/invoice/usage/details/UsageConsumableInArrearTierUnitDetail.java]: Nit: `BigDecimal.valueOf()`

File: invoice/src/main/java/org/killbill/billing/invoice/usage/ContiguousIntervalConsumableUsageInArrear.java [hunk 2]
```diff
@@ -114,29 +113,39 @@ protected UsageInArrearDetail getToBeBilledUsageDetails(final List<RolledUpUnit>
...
             for (final UsageConsumableInArrearTierUnitDetail curDetail : usageDetail.getTierDetails()) {
+
                 if (curDetail.getTierUnit().equals(unitType)) {
-                    result.add(curDetail);
+
+                    if (!resultMap.containsKey(curDetail.getTier())) {
+                        resultMap.put(curDetail.getTier(), curDetail);
+                    } else {
+                        final UsageConsumableInArrearTierUnitDetail perTierDetail = resultMap.get(curDetail.getTier());
+                        perTierDetail.updateQuantityAndAmount(perTierDetail.getQuantity() + curDetail.getQuantity());
```
Comment (pierre) [inline invoice/src/main/java/org/killbill/billing/invoice/usage/ContiguousIntervalConsumableUsageInArrear.java]: Nit: maybe `updateQuantityAndAmount(perTierDetail.getQuantity() + curDetail.getQuantity())` call could be simplified to `updateQuantityAndAmount(curDetail.getQuantity())` (i.e. the additional happens in `UsageConsumableInArrearTierUnitDetail`).

Comment (sbrossie) [reply invoice/src/main/java/org/killbill/billing/invoice/usage/ContiguousIntervalConsumableUsageInArrear.java]: Ok, will change the code.

File: invoice/src/main/java/org/killbill/billing/invoice/usage/details/UsageConsumableInArrearTierUnitDetail.java (hunk 1 above)
Comment (sbrossie) [reply invoice/src/main/java/org/killbill/billing/invoice/usage/details/UsageConsumableInArrearTierUnitDetail.java]: Ok will change.

**Comment by sbrossie** [comment]
Regarding test failure, are you sure you pulled the branch? I ran them 4 times and all of them succeed.

**Pull request merged**